src/Compilers/Core/Portable/Text/CompositeText.cs

```diff
@@ -182,6 +184,12 @@ internal static SourceText ToSourceText(ArrayBuilder<SourceText> segments, Sourc
                 ReduceSegmentCountIfNecessary(segments);
             }
 
+            // If there are multiple segments, ensure they all have non-zero length. This simplifies the CompositeTextLineInfo logic.
+            if (segments.Count > 1)
```

Comment:
nice.  consider adding check in constructor that all segments are non-zero length.